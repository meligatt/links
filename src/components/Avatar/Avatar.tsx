import * as React from "react";
import styled from "styled-components";

const AvatarStyled = styled.h3`
  border: 1px solid red;
`;

export const Avatar: React.FunctionComponent = (): JSX.Element => {
  return (
    <>
      <AvatarStyled> @obiwan </AvatarStyled>
    </>
  );
};
