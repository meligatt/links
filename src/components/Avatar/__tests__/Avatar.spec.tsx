import * as React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Avatar } from "../Avatar";

describe("Avatar", () => {
  test("renders image with default props", () => {
    const { container } = render(
      <Avatar src="https://meligatt.github.io/melissa-gattoni-frontend-dev.jpeg" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("renders image with passed props", () => {
    const { getByAltText } = render(
      <Avatar
        src="https://meligatt.github.io/melissa-gattoni-frontend-dev.jpeg"
        alt="mocked name"
        width={120}
        height={120}
      />
    );
    expect(getByAltText("mocked name")).toBeInTheDocument();
  });
});
