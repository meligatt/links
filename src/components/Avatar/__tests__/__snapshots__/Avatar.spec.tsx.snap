// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders image with default props 1`] = `
<img
  alt="user name"
  class="sc-AxjAm eBnlIx"
  height="64"
  src="https://meligatt.github.io/melissa-gattoni-frontend-dev.jpeg"
  width="64"
/>
`;
